lib/PermissionedRoute/PermissionedRoute.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { Route } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+import { IfPermission } from '@folio/stripes/core';
+
+import NoPermissionsMessage from './NoPermissionsMessage';
+
+const PermissionedRoute = ({ children, path, perm, returnLinkLabelId, returnLink }) => (
+  <IfPermission perm={perm}>
+    {({ hasPermission }) => (hasPermission
+      ? (
+        <Route
+          exact
+          path={path}
+          component={children}
```

Comment:
I'm not sure if this gonna work, children isn't always a component

Reply:
updated to component

Reply:
and how it will look in the place you call it? https://github.com/folio-org/ui-invoice/pull/375/files
for example ```
<Route
        path="/invoice/edit/:id"
```

Reply:
there is a reason why in the react-router examples they use only children https://reactrouter.com/web/example/auth-workflow